plugins/FitToContent/main.lua

```diff
@@ -0,0 +1,49 @@
+-- Register all Toolbar actions and intialize all UI stuff
+function initUi()
+    app.registerUi{["menu"] = "Fit page to layer", ["callback"] = "fit_to_layer"};
+    app.registerUi{["menu"] = "Fit page to selection", ["callback"] = "fit_to_selection"};
+end
+
+local function fit_to_layer_or_selection(type)
+    if type ~= "Selection" and type ~= "Layer" then
+        app.msgbox("Invalid type for fitting", {[1]="Ok"})
+        return
+    end
+
+    local strokes = app.getStrokes(type)
+    -- padd the whole region just a little bit. Might be a thing to configure in the future
+    local padding = {x=0, y=0}
+    -- search for the outermost strokes which span the rectangle of the to be moved area
+    -- take the stroke width, or if set the pressure, into account hereby
+    local mima    = {minX=100000, maxX=0, minY=100000, maxY=0}
+    for _,stroke in ipairs(strokes) do
+        for i,x in ipairs(stroke.x) do
+            mima.maxX = math.max(mima.maxX, x + (stroke.pressure and stroke.pressure[i] or stroke.width/2))
+            mima.minX = math.min(mima.minX, x - (stroke.pressure and stroke.pressure[i] or stroke.width/2))
+        end
+        for i,y in ipairs(stroke.y) do
+            mima.maxY = math.max(mima.maxY, y + (stroke.pressure and stroke.pressure[i] or stroke.width/2))
+            mima.minY = math.min(mima.minY, y - (stroke.pressure and stroke.pressure[i] or stroke.width/2))
```

Comment:
The pressure values must also be divided by 2. Moreover performance would be higher if `stroke.pressure` was only checked once for every stroke and not for every point of the stroke.

Reply:
Yes, performance would be higher this way (and it's right that performance in these loops is critical). But in fact we need to check for every point if the pressure value should be taken or the width.

The solution that comes to my mind (apart from duplicating the inner loops) is to work with functions like
```lua
local function padd_with_pressure(stroke, i)
    return stroke.pressure[i]
end
local function padd_without_pressure(stroke)
    return stroke.width
end
```
and then set a variable to this function before going into the inner loops to either of these functions (depending if pressure is present or not).
But I'm not sure, if this approach really increases the performance (and benchmarking is not that easy in this case, I think).

Any ideas on the performance/pressure checking part?

Reply:
I would simply duplicate the inner loops.